aws/resource_aws_codebuild_project.go

```diff
@@ -659,6 +679,22 @@ func expandProjectEnvironment(d *schema.ResourceData) *codebuild.ProjectEnvironm
 		projectEnv.ImagePullCredentialsType = aws.String(v.(string))
 	}
 
+	if v := envConfig["registry_credential"]; v != nil {
+		config := v.(*schema.Set).List()[0].(map[string]interface{})
+
+		projectRegistryCredential := &codebuild.RegistryCredential{}
+
+		if v := config["credential"].(string); v != "" {
+			projectRegistryCredential.Credential = &v
+		}
+
+		if v := config["credential_provider"].(string); v != "" {
+			projectRegistryCredential.CredentialProvider = &v
```

Comment:
Similar change here:

```suggestion
			projectRegistryCredential.CredentialProvider = aws.String(v.(string))
```